apis/python/node/dora/cuda.py

```diff
@@ -24,75 +30,65 @@ def torch_to_ipc_buffer(tensor: torch.TensorType) -> tuple[pa.array, dict]:
     ```
     """
     device_arr = to_device(tensor)
-    cuda_buf = pa.cuda.CudaBuffer.from_numba(device_arr.gpu_data)
-    handle_buffer = cuda_buf.export_for_ipc().serialize()
+    ipch = get_context().get_ipc_handle(device_arr.gpu_data)
+    _, handle, size, source_info, offset = ipch.__reduce__()[1]
     metadata = {
         "shape": device_arr.shape,
         "strides": device_arr.strides,
         "dtype": device_arr.dtype.str,
+        "size": size,
+        "offset": offset,
+        "source_info": json.dumps(source_info),
     }
-    return pa.array(handle_buffer, type=pa.uint8()), metadata
+    return pa.array(handle, pa.int8()), metadata
```

Comment:
This should be uint8 right?
```suggestion
    return pa.array(handle, pa.uint8()), metadata
```

Reply:
`handle` is actually a int8 array (ranged from `-128` to `127`), since its underlying c type is `char`